hooks/Useregister.ts

```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+import { useRouter } from "next/router";
+import { useState } from "react";
+import { RegisterProps } from "../types";
+
+export const useRegister = () => {
+  const [errorText, setErrorText] = useState("");
+  const router = useRouter();
+  
+  const register = async ({ name, email, password }: RegisterProps) => {
+    try {
+      axios
+        .post(`${process.env.url}/register`, {
+          name: name,
+          email: email,
+          password: password,
+        })
+        .then((res) => {
+          if (res.status === 200 || 201) {
+            router.push("/");
+            localStorage.setItem("token", res.data.accessToken);
+            localStorage.setItem("name", res.data.user.name);
+          }
+        })
+        .catch((err) => {
+          setErrorText(err.message);
+          setTimeout(() => {
+            setErrorText("");
+          }, 2000);
+        });
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return { register, errorText };
```

Comment:
and think, maybe we can use one hook for login and register? code of both seems very-very similar